### PR TITLE
Add TouchBehavior and ImageTouchBehavior documentation

### PR DESCRIPTION
**Pull request by bijington**
Provides documentation for https://github.com/CommunityToolkit/Maui/pull/1673

### CODE REVIEW

File: docs/maui/behaviors/image-touch-behavior.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+# ImageTouchBehavior
+
+The `ImageTouchBehavior` extends the `TouchBehavior` by providing the ability to customize `Image` elements based on touch, mouse click and hover events. The `ImageTouchBehavior` implementation makes it possible to customize the `Source` property of the `Image` element that it is attached to.
+
+> [!NOTE]
+> For further customization options in the `ImageTouchBehavior` please refer to the [`TouchBehavior`](./touch-behavior.md) implementation that this `Behavior` extends.
+
+## Syntax
+
+The following examples show how to add the `ImageTouchBehavior` to an `Image` and toggle between the `DefaultImageSource` and the `PressedImageSource` when the users interacts with the image by either pressing their finger or clicking with the mouse on the `Image`.
```
Comment (brminnick) [inline docs/maui/behaviors/image-touch-behavior.md:17]: Let's include `HoveredImageSource` too

File: docs/maui/behaviors/image-touch-behavior.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    x:Class="CommunityToolkit.Maui.Sample.Pages.Behaviors.TouchBehaviorPage"
+    x:Name="Page">
+
+    <Image HeightRequest="100" WidthRequest="100">
+        <Image.Behaviors>
+            <toolkit:ImageTouchBehavior
+                Command="{Binding Source={x:Reference Page}, Path=BindingContext.IncreaseTouchCountCommand}"
+                DefaultImageSource="button.png"
+                PressedImageSource="button_pressed.png" />
+        </Image.Behaviors>
```
Comment (brminnick) [inline docs/maui/behaviors/image-touch-behavior.md:41]:
```suggestion
                PressedImageSource="button_pressed.png"
                HoveredImageSource="button_hovered.png" />
        </Image.Behaviors>
```

File: docs/maui/behaviors/image-touch-behavior.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+}
+```
+
+## Examples
+
+You can find an example of this behavior in action in the [.NET MAUI Community Toolkit Sample Application](https://github.com/CommunityToolkit/Maui/blob/main/samples/CommunityToolkit.Maui.Sample/Pages/Behaviors/TouchBehaviorPage.xaml).
+
+## API
+
+You can find the source code for `ImageTouchBehavior` over on the [.NET MAUI Community Toolkit GitHub repository](https://github.com/CommunityToolkit/Maui/blob/main/src/CommunityToolkit.Maui/Behaviors/ImageTouchBehavior.shared.cs).
```
Comment (brminnick) [inline docs/maui/behaviors/image-touch-behavior.md:107]: Reminder to add the public APIs (minus the public APIs in TouchBehavior.shared.cs)

File: docs/maui/behaviors/touch-behavior.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+</HorizontalStackLayout>
+```
+
+## Examples
+
+You can find an example of this behavior in action in the [.NET MAUI Community Toolkit Sample Application](https://github.com/CommunityToolkit/Maui/blob/main/samples/CommunityToolkit.Maui.Sample/Pages/Behaviors/TouchBehaviorPage.xaml).
+
+## API
+
+You can find the source code for `TouchBehavior` over on the [.NET MAUI Community Toolkit GitHub repository](https://github.com/CommunityToolkit/Maui/blob/main/src/CommunityToolkit.Maui/Behaviors/TouchnBehavior.shared.cs).
```
Comment (brminnick) [inline docs/maui/behaviors/touch-behavior.md:211]: Reminder to add the massive amount of TouchBehavior public APIs

**Review by brminnick — COMMENTED** [review]
Looking good! I've only had a chance to review ImageTouchBehavior. 

It looks like the biggest lift for this PR will be adding all of the public TouchBehavior APIs.

**Review by jfversluis — APPROVED** [review]

**Pull request merged**